### PR TITLE
PEAR-762 - Put context id in local storage

### PR DESCRIPTION
**Pull request by amlehman**
## Checklist

- [x] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [x] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

File: packages/core/src/listeners.ts [hunk 1]
```diff
@@ -85,3 +87,39 @@ startCoreListening({
...
+    cohortApiSlice.endpoints.getCohortsByContextId.matchPending,
+    cohortApiSlice.endpoints.getCohortById.matchPending,
+    cohortApiSlice.endpoints.addCohort.matchPending,
+    cohortApiSlice.endpoints.updateCohort.matchPending,
+    cohortApiSlice.endpoints.deleteCohort.matchPending,
+  ),
+  effect: async () => {
+    // If cookie has been deleted, restore it from local storage
+    if (!Cookies.get("gdc_context_id")) {
+      const contextId = localStorage.getItem("gdc_context_id");
```
Comment (paribartandhakal) [inline packages/core/src/listeners.ts:119]: Will there be a case when both the localStorage and Cookies wont have this? If so what happens.

Comment (amlehman) [reply packages/core/src/listeners.ts:119]: I believe the backend will create a context id and attach the set-cookie header if the request doesn't have a context id.

**Review by paribartandhakal — DISMISSED** [review]
🥇

**Comment by wteouchicago** [comment]
I want to understand something / something doesn't feel right:
- incognito
- verified localstorage and cookie doesn't have the context_id
- make a cohort, save, e.g. "Eye"
- verified localstorage and cookie now have the context_id
- delete the context_id from cookie, then nuked the whole cookie for good measure
- tried saving -- ran into cohort saving error
- tried refreshing -- lost my "Eye" cohort
- but the cookie now has the context_id

I would have thought that now that the cookie has the context_id, I should be able to retrieve my "Eye" cohort that was saved and also be able to modify "Eye".

I then made another cohort and saved that. Now I have 2 context_ids in the cookie. The "old"/"first" one has an expiration/max-age set to "Session".

**Comment by amlehman** [comment]
Hi @wteouchicago, I see the issue you're describing. Not sure how it was working for me before but I've pushed a change that should fix the cookie not getting added to the requests. Could you test again when you get a chance?

**Comment by wteouchicago** [comment]
very nice -- everything is working as expected and smoothly. 👍 I tried variations of deleting cookies, deleting the context_id from cookie or from localstorage and nothing was disrupted.

**Review by craigrbarnes — APPROVED** [review]
Works for me as well

**Pull request merged**